contrib/vagrant/provision-util.sh

```diff
@@ -179,6 +182,133 @@ os::provision::get-network-plugin() {
   echo "${plugin}"
 }
 
+os::provision::install-networking() {
+  local plugin=$1
+  local master_ip=$2
+  local deployed_root=$3
+  local config_root=$4
+  local is_master=${5:-false}
+
+  echo "Configuring networking"
+
+  if [[ "${plugin}" =~ redhat/ ]]; then
+    os::provision::install-sdn "${deployed_root}"
+  elif [[ "${plugin}" = "flannel" ]]; then
+    os::provision::install-flannel "${master_ip}" "${deployed_root}" \
+"${config_root}" "${is_master}"
+  else
+    >&2 echo "Unable to deploy network plugin ${plugin}"
+    exit 1
+  fi
+}
+
+os::provision::install-flannel() {
+  local master_ip=$1
+  local deployed_root=$2
+  local config_root=$3
+  local is_master=$4
+
+  if os::provision::in-container; then
+    local if_name=eth0
+  else
+    yum install -y flannel
+
+    local conf_path=/etc/sysconfig/network-scripts/
+    local if_to_edit=$(find ${conf_path}ifcfg-* | xargs grep -l VAGRANT-BEGIN)
+    local if_name=`echo ${if_to_edit} | awk -F- '{ print $3 }'`
+  fi
+
+  local cert_path="${config_root}/openshift.local.config/master"
+  local ca_file="${cert_path}/ca.crt"
+  local cert_file="${cert_path}/master.etcd-client.crt"
+  local key_file="${cert_path}/master.etcd-client.key"
+  local etcd_url="https://${master_ip}:4001"
+  local etcd_key="/atomic.io/network"
+
+  if [ "${is_master}" = "true" ]; then
+    local mode="listen"
+
+    cat <<EOF > /etc/flannel-config.json
+{
+  "Network": "10.1.0.0/16",
+  "SubnetLen": 24,
+  "Backend": {
+      "Type": "udp",
+      "Port": 8285
+   }
+}
+EOF
+
+    local etcdctl_cmd="${deployed_root}/_tools/etcd/bin/etcdctl -C ${etcd_url}
+--ca-file ${ca_file} --cert-file ${cert_file} --key-file ${key_file}"
+
+    local msg="etcd daemon to become available"
+    local condition="os::provision::check-etcd ${etcdctl_cmd}"
+    os::provision::wait-for-condition "${msg}" "${condition}"
+
+    # Import default configuration into etcd
+    ${etcdctl_cmd} set "${etcd_key}/config" < /etc/flannel-config.json \
+> /dev/null
+
+  else
+    local mode="remote"
+  fi
+
+  local daemon_opts="-iface=${if_name} -ip-masq=false -v=5 \
+-etcd-cafile=${ca_file} -etcd-certfile=${cert_file} -etcd-keyfile=${key_file} \
+-${mode}=${master_ip}:8080"
+
```

Comment:
The node flannels don't need any of the etcd config stuff, since they never talk to etcd directly.  I think I probably gave you the wrong node command-line originally since in my cheat-sheet I have it wrong too.  But it really only need to be:

/usr/bin/flanneld -ip-masq=false -v=5 -listen=10.245.2.2:8080